gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp

```diff
@@ -244,49 +243,73 @@ void GazeboRosControlPlugin::Load(gazebo::physics::ModelPtr parent, sdf::Element
     return;
   }
 
-  // Load the RobotHWSim abstraction to interface the controllers with the gazebo model
+  std::unique_ptr<hardware_interface::ResourceManager> resourceManager_ =
+    std::make_unique<hardware_interface::ResourceManager>();
+
   try {
+    // We cannot use (for now) the <ros2_control> tag in the URDF, we need to call it manually.
     impl_->robot_hw_sim_loader_.reset(
-      new pluginlib::ClassLoader<gazebo_ros2_control::RobotHWSim>(
+      new pluginlib::ClassLoader<gazebo_ros2_control::GazeboSystemInterface>(
         "gazebo_ros2_control",
-        "gazebo_ros2_control::RobotHWSim"));
+        "gazebo_ros2_control::GazeboSystemInterface"));
 
-    impl_->robot_hw_sim_ = impl_->robot_hw_sim_loader_->createSharedInstance(
-      impl_->robot_hw_sim_type_str_);
+    auto gazeboSystem = std::unique_ptr<gazebo_ros2_control::GazeboSystemInterface>(
+      impl_->robot_hw_sim_loader_->createUnmanagedInstance(impl_->robot_hw_sim_type_str_));
 
     urdf::Model urdf_model;
     const urdf::Model * const urdf_model_ptr =
       urdf_model.initString(urdf_string) ? &urdf_model : NULL;
 
     rclcpp::Node::SharedPtr node_ros2 = std::dynamic_pointer_cast<rclcpp::Node>(impl_->model_nh_);
-    if (!impl_->robot_hw_sim_->initSim(
-        impl_->model_nh_->get_namespace(), node_ros2, impl_->parent_model_, urdf_model_ptr,
-        impl_->transmissions_))
+    if (!gazeboSystem->initSim(
+        node_ros2,
+        impl_->parent_model_,
+        urdf_model_ptr,
+        impl_->transmissions_,
+        sdf))
     {
       RCLCPP_FATAL(
         impl_->model_nh_->get_logger(), "Could not initialize robot simulation interface");
       return;
     }
 
+    std::vector<hardware_interface::LoanedStateInterface> loaned_state_ifs;
+    std::vector<hardware_interface::StateInterface> state_ifs;
+    state_ifs = gazeboSystem->export_state_interfaces();
+    for (auto state : state_ifs) {
+      loaned_state_ifs.emplace_back(hardware_interface::LoanedStateInterface(state));
+    }
+
+    resourceManager_->import_component(std::move(gazeboSystem));
```

Comment:
why are you manually exporting the state interfaces here? `import_component` should be all you need to use.